File: app/src/main/java/sfedu/physics/dmitriy/githubapiresttestproject/utils/QueryUtils.java
package sfedu.physics.dmitriy.githubapiresttestproject.utils;

public class QueryUtils {
    public static String createQueryByLocationAndLanguage(String userLocation, String userProgrammingLanguage) {
        StringBuilder query = new StringBuilder();
        if (!userLocation.isEmpty())
            query.append(String.format("location:%s+", userLocation));
        if (!userProgrammingLanguage.isEmpty())
            query.append(String.format("language:%s", userProgrammingLanguage));

        return query.toString();
    }
}

File: app/src/main/java/sfedu/physics/dmitriy/githubapiresttestproject/utils/QueryUtils.kt
package sfedu.physics.dmitriy.githubapiresttestproject.utils

class QueryUtils {
    companion object {
        fun  createQueryByLocationAndLanguage(userLocation: String, userProgrammingLanguage: String): String {
            val query = StringBuilder()
            if (!userLocation.isEmpty())
                query.append(String.format("location:%s+", userLocation))
            if (!userProgrammingLanguage.isEmpty())
                query.append(String.format("language:%s", userProgrammingLanguage))

            return query.toString()
        }
    }
}
